### PR TITLE
New: Check node version based on package.json

### PR DESCRIPTION
**Pull request by stramel**
- [x] CHANGELOG.md has been updated

### CODE REVIEW

File: src/polymer-cli.ts [hunk 1]
```diff
@@ -12,6 +12,9 @@
  * http://polymer.github.io/PATENTS.txt
  */
 
+// Bail out if invalid node version
+require('please-upgrade-node')(require('../package.json'));
```
Comment (justinfagnani) [inline src/polymer-cli.ts]: We already have similar logic here: https://github.com/Polymer/polymer-cli/blob/master/bin/polymer.js#L27

Also, I'd be hesitant to put statements before import declarations, as that's not actually spec-compliant.

Comment (stramel) [inline src/polymer-cli.ts]: Doh! I definitely didn't look here.

I put this in the wrong place apparently. The require before import is to ensure that older versions that don't support `import`. By placing it in bin/polymer.js it is technically before all the other imports. 

I will modify this to do the bin/polymer.js

File: bin/polymer.js [hunk 2]
```diff
@@ -22,11 +22,12 @@ process.title = 'polymer';
...
 // NOTE 04-21-2017: Confirmed "semver" supports Node versions as low as 0.10
 var semver = require('semver');
+var version = require('../package.json').engines.node;
 
 // Exit early if the user's node version is too low.
-if (!semver.satisfies(process.version, '>=6')) {
+if (!semver.satisfies(process.version, version)) {
   console.log(
-      'Polymer CLI requires at least Node v6. ' +
+      'Polymer CLI requires at least Node v' + verion.replace(/[^\d\.]*/, '') + '. '
```
Comment (abdonrd) [inline bin/polymer.js]: Typo: `verion` > `version`.

Comment (justinfagnani) [inline bin/polymer.js]: Missing `+` at the end of the line.

Comment (justinfagnani) [inline bin/polymer.js]: Can you store the `version.replace` result in a variable and comment what it's doing?

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,6 +1,7 @@
 # Changelog
 
 ## Unreleased
+- Drives node version check from the package.json
```
Comment (TimvdLippe) [inline CHANGELOG.md]: Derives 😉

Comment (stramel) [reply CHANGELOG.md]: Updated

**Comment by stramel** [comment]
@justinfagnani @FredKSchott Following up on this, I have resolved the changelog conflicts

**Comment by stramel** [comment]
Thanks @justinfagnani looks like the build failed on `lint` integration test. Not sure if this is a flake or a change someone else introduced.

**Comment by stramel** [comment]
Any chance we can get this merged? I just rebased again. I also regenerated the package-lock.json since I had conflicts in there and none seemed to be any breaking changes.

**Comment by justinfagnani** [comment]
@stramel merged!

**Pull request merged**